router/keeper/keeper.go

```diff
@@ -68,27 +88,66 @@ func (k Keeper) ForwardTransferPacket(ctx sdk.Context, parsedReceiver *parser.Pa
 
 	// pay fees
 	if feeAmount.IsPositive() {
-		err = k.distrKeeper.FundCommunityPool(ctx, feeCoins, parsedReceiver.HostAccAddr)
+		hostAccAddr, err := sdk.AccAddressFromBech32(parsedReceiver.HostAccAddr)
 		if err != nil {
+			return err
+		}
+		err = k.distrKeeper.FundCommunityPool(ctx, feeCoins, hostAccAddr)
+		if err != nil {
+			k.Logger(ctx).Error("packetForwardMiddleware error funding community pool",
+				"error", err,
+			)
 			return sdkerrors.Wrapf(sdkerrors.ErrInsufficientFunds, err.Error())
 		}
 	}
 
 	// send tokens to destination
-	err = k.transferKeeper.SendTransfer(
-		ctx,
-		parsedReceiver.Port,
-		parsedReceiver.Channel,
-		packetCoin,
-		parsedReceiver.HostAccAddr,
-		parsedReceiver.Destination,
-		DefaultTransferPacketTimeoutHeight,
-		DefaultTransferPacketTimeoutTimestamp+uint64(ctx.BlockTime().UnixNano()),
+	packet, err := k.transferKeeper.Transfer(
+		sdk.WrapSDKContext(ctx),
+		transfertypes.NewMsgTransfer(
+			parsedReceiver.Port,
+			parsedReceiver.Channel,
+			packetCoin,
+			parsedReceiver.HostAccAddr,
+			parsedReceiver.Destination,
+			DefaultTransferPacketTimeoutHeight,
+			uint64(ctx.BlockTime().UnixNano())+uint64(timeout.Nanoseconds()),
+		),
 	)
 	if err != nil {
+		k.Logger(ctx).Error("packetForwardMiddleware SendPacketTransfer error",
+			"error", err,
+			"amount", packetCoin.Amount.String(),
+			"denom", packetCoin.Denom,
+			"sender", parsedReceiver.HostAccAddr,
+			"receiver", parsedReceiver.Destination,
+			"port", parsedReceiver.Port,
+			"channel", parsedReceiver.Channel,
+		)
 		return sdkerrors.Wrapf(sdkerrors.ErrInsufficientFunds, err.Error())
 	}
 
+	// Store the following information in keeper:
+	// key - information about forwarded packet: src_channel (parsedReceiver.Channel), src_port (parsedReceiver.Port), sequence
```

Comment:
We should probably have this comment in the protobufs and other places where this is used (genesis)